src/hooks/useGatheringButtonState.ts

```
@@ -0,0 +1,67 @@
+  minParticipants,
+}: UseGatheringButtonStateParams) {
+  // 참가 여부 확인
+  const joined = joinedGatherings?.some(g => g.id === Number(gatheringId)) ?? false;
+
+  // 참가자 수 계산
+  const currentParticipantCount = participantsData?.length ?? gathering?.participantCount ?? 0;
+  const capacity = gathering?.capacity ?? 20;
+
+  // 정원 초과 여부
+  const isFull = currentParticipantCount >= capacity;
+  const capacity = gathering?.capacity ?? 20;
```

Biome (2.1.2)

[error] This variable is used before its declaration.

The variable is declared here:


(lint/correctness/noInvalidUseBeforeDeclaration)